src/Client.cpp

```diff
@@ -8,28 +10,36 @@
 #include "libursa/ZHelpers.h"
 #include "spdlog/spdlog.h"
 
-static void wait_sec() {
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+bool UrsaClient::wait_sec() {
```

Comment:
This looks pretty hacky, but I don't know why your threads kept hanging.

Just in case, are you sure this solves the problem, or just makes it less visible? (i.e. now the time window for a bug is 10ms instead of 1000ms).

Also, how CPU intensive is this? If ursacli uses something like 10% cpu by nooping it'd be pretty bad.

Reply:
it's not related to any bug, imagine that you run
```
ursacli -c status
```
and the status thread loops for 1 second, so you get the output immediately but the program is waiting additional 1 second before the status thread realizes that it should terminate

Reply:
How did we solve this in python? I don't recall any threads there.

IIRC we just had a while() loop waiting for response with 1s timeout, and printing a progress bar every 1s. Maybe this will solve all these concurrency and threading problems?

Reply:
Then we should probably do some short-time polling.

Reply:
I mean, in this case it works like this:

```
while True:
    x = recv(timeout=1s)
    if timeout():
        print_progress()
        continue
    else:
        print_result(x)
        break
```

Reply:
I would skip async pooling for now, as it would require a huge rework of the client (introducing new bugs), while at the moment we've (with @chivay) realized how to patch the existing ones :P